standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -855,6 +868,20 @@ boolean getPartitionsByExpr(String catName, String dbName, String tblName,
       List<Partition> result, GetPartitionsArgs args)
       throws TException;
 
+  /**
+   * Prune the input partition names that match the expressions.
+   * @param catName catalog name
+   * @param dbName database name
+   * @param tblName table name
+   * @param result input list of partition names
+   * @param args additional arguments for pruning
+   * @return true if the result contains unknown partition names.
+   * @throws MetaException error executing the expression
+   */
+  boolean prunePartitionNamesByExpr(String catName, String dbName, String tblName,
```

Comment:
why we need this method in `RawStore`, I don't see it being called in the `HMSHandler`

Reply:
change to `private` in `ObjectStore`.